server/src/test/java/org/candlepin/async/tasks/UndoImportsJobTest.java

```diff
@@ -0,0 +1,235 @@
+/**
+ * Copyright (c) 2009 - 2012 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.async.tasks;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.anyBoolean;
+import static org.mockito.Mockito.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.candlepin.async.JobConfig;
+import org.candlepin.async.JobExecutionContext;
+import org.candlepin.async.JobExecutionException;
+import org.candlepin.controller.CandlepinPoolManager;
+import org.candlepin.controller.Refresher;
+import org.candlepin.model.Consumer;
+import org.candlepin.model.ConsumerType;
+import org.candlepin.model.Entitlement;
+import org.candlepin.model.ExporterMetadata;
+import org.candlepin.model.ExporterMetadataCurator;
+import org.candlepin.model.ImportRecord;
+import org.candlepin.model.ImportRecordCurator;
+import org.candlepin.model.Owner;
+import org.candlepin.model.OwnerCurator;
+import org.candlepin.model.Pool;
+import org.candlepin.model.Pool.PoolType;
+import org.candlepin.model.Product;
+import org.candlepin.model.UeberCertificate;
+import org.candlepin.model.UeberCertificateGenerator;
+import org.candlepin.model.UpstreamConsumer;
+import org.candlepin.service.OwnerServiceAdapter;
+import org.candlepin.service.SubscriptionServiceAdapter;
+import org.candlepin.test.DatabaseTestFixture;
+import org.candlepin.test.TestUtil;
+
+import com.google.inject.Inject;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+import org.xnap.commons.i18n.I18n;
+import org.xnap.commons.i18n.I18nFactory;
+
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+
+
+
+/**
+ * UndoImportsJobTest
+ */
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+public class UndoImportsJobTest extends DatabaseTestFixture {
```

Comment:
I noticed that the old `UndoImportsJobTest` has some re-fire scenario test cases. Those were basically testing the scenarios that the job itself was handling with catch clauses in the past, but then that logic then moved to the general KingpinJob re-fire logic. Do we need to bring that KingpinJob logic in the new framework (e.g. making sure we re-try when we find lock exceptions or runtime exceptions that wrap SQLExceptions), since it seems it is desirable?

Reply:
In the new framework, all jobs will refire on unexpected/uncaptured exceptions as the default behavior. We only need special configuration if we *don't* want that behavior.

Reply:
As a side note, this is something that we can add to the job config if we need to fiddle with the default behavior for jobs wrt how they handle uncaught exceptions.